### PR TITLE
@accounts/types : add a types package which contains common types

### PR DESCRIPTION
**Pull request by Aetherall**

### CODE REVIEW

**Comment by pradel** [comment]
Why do we need a new types package, can this go to common instead?

**Comment by Aetherall** [comment]
In fact the common package actually is a dependency in some packages and a devDependency in others
the common package exports utilities that will no longer need to be shared across the suite in the future versions of Accounts-Js

That's why I am creating @accounts/types and @accounts/error

@accounts/error will be the dependency
@accounts/types will be the devDependency

It keeps the relation between packages simple

File: packages/types/package.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    },
+    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx)$",
+    "moduleFileExtensions": [
+      "ts",
+      "js"
+    ]
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/js-accounts/accounts/tree/master/packages/server"
```
Comment (davidyaha) [inline packages/types/package.json]: https://github.com/js-accounts/accounts/tree/master/packages/types

Comment (Aetherall) [reply packages/types/package.json]: Oh yes I am changing that

**Comment by Aetherall** [comment]
There is also DatabaseInterface because it makes more sense to have it here
I am adding the readme

**Comment by davidyaha** [comment]
@Aetherall A user with TS in his project needs to add this package to devDependencies as well?

**Comment by Aetherall** [comment]
@davidyaha only if he is using those types I suppose...

File: packages/types/readme.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Accounts Types
+### @accounts/types
+
+This package is a devDependency for most packages of the suite.
+
+It contains the types used across the Accouts Suite.
+
+#### Usage : This package is used internally by the suite. It's not needed to install it
```
Comment (davidyaha) [inline packages/types/readme.md]: OK, so maybe remove that.

Comment (Aetherall) [reply packages/types/readme.md]: Done

File: packages/types/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+export { EmailRecord } from './types/email-record'
+
+// ? Operation Results
+export { LoginResult as LoginReturnType } from './types/login-result'
+export { ImpersonationResult as ImpersonateReturnType } from './types/impersonation-result'
+
+
+// ? Operation Parameters
```
Comment (pradel) [inline packages/types/src/index.ts]: we will need to figure this before merging no?

File: packages/types/src/types/user.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { EmailRecord } from './email-record';
+
+export interface User {
+  username?: string;
+  email?: string;
```
Comment (pradel) [inline packages/types/src/types/user.ts]: is this line accurate?

Comment (Aetherall) [reply packages/types/src/types/user.ts]: In fact I just copied the types and renamed it

Comment (pradel) [reply packages/types/src/types/user.ts]: Yeah I now but was wondering if this is used or not ^^

Comment (pradel) [reply packages/types/src/types/user.ts]: I think it should not be used no?

File: packages/types/src/index.ts (hunk 3 above)
Comment (Aetherall) [reply packages/types/src/index.ts]: Yes actually I want to discuss it ASAP

File: packages/types/src/types/user.ts (hunk 4 above)
Comment (Aetherall) [reply packages/types/src/types/user.ts]: you mean the email field?

Comment (pradel) [reply packages/types/src/types/user.ts]: email string field

Comment (Aetherall) [reply packages/types/src/types/user.ts]: I agree, this was on the UserObjectType interface

File: packages/types/src/index.ts (hunk 3 above)
Comment (pradel) [reply packages/types/src/index.ts]: So this will be used on the password server and password client right?

Comment (Aetherall) [reply packages/types/src/index.ts]: the @accounts/types package will be used on pretty much every @accounts package, as a devDependency

Comment (pradel) [reply packages/types/src/index.ts]: I mean the type

Comment (Aetherall) [reply packages/types/src/index.ts]: Yes it is

Comment (davidyaha) [reply packages/types/src/index.ts]: What is the idea of Operation Parameters?

Comment (Aetherall) [reply packages/types/src/index.ts]: For several operations, the function expect one object as parameters which contains all the operation needs to be complete

**Pull request merged**